### DISCUSSION TITLE
利用 Vivaldi + 微信读书 + 文件传输助手 + 简悦让你脱离微信体系

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3404] [服务]
> 如果你像我一样，希望尽量降低微信对你的影响，但不想脱离微信体系（尤其是公众号）的话，可以试试下面的方式。

# 起因

前几天，一个简悦用户问我：

> 能否使用简悦方便的将看的内容转移到微信读书？

大多数经常使用微信的用户都知道可以在微信读书查看公众号并且可以导入多种格式的文件到微信读书，但由于众所周知的原因导致这个流程并不顺畅，也就是很难通过自动化方案来顺畅的使用它。

经过几天的「探索和实践」，找到了一个比较好的方案，基本上可以满足这位用户的需求，同时也能有效的降低对微信的依赖。

# 工作流

> 将文件传输助手与微信读书网页版添加到 Vivaldi 的侧栏，利用侧栏可以跟网页「平级」的优势可以减少切换 Tab 或窗口的频率，从而增加这个流程的顺畅度。

1. 利用简悦导出任意文章为 `.docx` 并传输到微信读书，方便在手机端查看你需要的内容。
2. 利用微信读书可以订阅公众号的功能，暂时摆脱使用微信阅读公众号。
3. 利用文件传输助手，可以方便的将任意 URL 通过 PC 端打开并使用简悦进行标注、生成快照。
3. 微信读书可以方便的将公众号分享到快捷指令，从而直接保存到简悦的稍后读（包括快照）。

# 视频演示

- 利用文件传输助手，可以方便的将任意 URL 通过 PC 端打开并使用简悦进行标注、生成快照

  https://user-images.githubusercontent.com/81074/153575666-b661b961-33df-49e2-a4bc-a725da1fe752.mp4

- 利用简悦导出任意文章为 `.docx` 并传输到微信读书，方便在手机端查看你需要的内容

  https://user-images.githubusercontent.com/81074/153575708-130bba83-c6ad-4706-9353-19af3516e5fc.mp4
  
  https://user-images.githubusercontent.com/81074/153563725-f5a88723-c432-4d9a-be5b-6bb4bf916e94.mp4

# 配置

## 前置知识

使用此功能需要事先配置如下内容：

- [简悦 · 高级账户](https://simpread.pro/price.html)
- 配置 [同步助手](https://www.yuque.com/kenshin/simpread/pwpnsx)
- 配置 [增强导出](https://www.yuque.com/kenshin/simpread/bwcd6n)

## Vivaldi 侧栏

- 将 [微信读书](https://weread.qq.com/) 加入到侧栏，如果首次在侧栏使用微信读书的话，需要登录

  ![image-20220211162146646](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644567706.png)

- 调整侧栏各选项为下图所示

  ![image-20220211162421032](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644567861.png)

- 因为无法直接在侧栏中使用 https://weread.qq.com/web/upload ，所以需要再建立一个新的网页面板，并添加 https://weread.qq.com/web/upload

  ![image-20220211162048476](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644567648.png)

- 登录成功后类似下图的效果，并删除上一个

  ![image-20220211163105838](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644568265.png)

- 可以在设置中调整侧栏面板的各项内容，打开偏好设置

  ![image-20220211163209942](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644568329.png)

- 将 [文件传输助手](https://filehelper.weixin.qq.com/) 按上面一样的方式添加到侧栏（仅添加一次即可），添加后登录，如下图

  ![image-20220211163439463](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644568479.png)

利用 Vivaldi 的优势：

- 侧栏可以跟标签页平级，省去了切换多个 Tab 的操作。

- 不需要侧栏时可以将其收回
  
  https://user-images.githubusercontent.com/81074/153576861-4db0db1f-9c1a-4d98-a02f-eedbd7af00a0.mp4

## 简悦端设置

简悦有着丰富的导出功能，几乎支持市面上几乎全部的导出方案，而微信读书的导入也基本上涵盖了大不多数文档格式如：`.epub,.txt,.pdf,.doc,.docx,.mobi,.azw3` 等，为了得到更好的方案，我做了一个简单的测试，分别使用简悦导出：

- PDF
- Epub
- docx

对比后效果最好的 `.docx` 格式，不仅可以包含完整的图片格式，也包含了精准、完整的目录。（过程省略）

**结论：**将阅读模式导出为 `.docx` 再传入到微信读书中，因此需要事先配置 **增强导出**。

### 安装 Pandoc

简悦的 [增强导出](https://www.yuque.com/kenshin/simpread/bwcd6n) 部分功能基于 Pandoc，需要下载安装 Pandoc，[下载地址](https://github.com/jgm/pandoc/releases) 找到对应操作系统的版本即可，安装很简单，一路 Next 即可。

安装后在命令行下输入 `pandoc -v` 看到正常的输出则证明配置成功。

#### 验证（Windows）

一般 Windows 的安装无需设置，只要 **确保全局可以调用 pandoc 命令**，安装后直接就可以使用。

#### 验证（Mac）

类似 Windows，确保 pandoc 在你的 `/usr/local/bin/` 目录下存在，同时也确保此目录在 `$PATH` 下。

> 只要安装了 Pandoc 即可，余下的事情简悦同步助手就可以自动帮你完成，接下来需要配置自定义导出。

### 简悦扩展端

进入简悦 → 选项页 → 服务 → 定制导出，复制下面的代码粘

```
{"name":"Markdown 转换 docx", "in":"markdown", "out":"docx", "icon":"<i class=\"fas fa-file-word\"></i>"}
{"name":"HTML 转换 docx", "in":"html", "out":"docx"}
```

贴到下图所示位置即可（留意 `name` 字段的名称）

![image-20220211181436632](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644574476.png)

至此，已完成全部配置。

# 如何使用

> 支持手动导出或者自动化导出方案。

## 手动导出

打开任意页面 e.g. https://sspai.com/post/71200 进入阅读模式，右下角触发器 → 动作，根据下图所示点击使用。

![image-20220211181738604](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644574658.png)

注意：这里也可以改用 `Markdown 转换 docx`

![img](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644574744.png)

## 自动化导出

> 配合简悦的 [自动化](http://ksria.com/simpread/docs/#/自动化)，可在加入稍后读后，自动生成 `.docx` 文件。

- 进入选项页 → 服务 → 自动化，新建一个自动化方案

  ![image-20220211184925703](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644576565.png)

- 并按下图所示设置（主要是将 `HTML 转换 docx` 加入到自动化方案）

  ![image-20220211184739434](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644576459.png)

- 在标注时，将  `HTML 转换 docx`  加入到标注的自动化（可选）

  ![image-20220211185128290](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20220211_1644576688.png)

至此，可以使用快捷键 `d d` 加入稍后读时，就会触发自动化生成 `.docx` 文件

https://user-images.githubusercontent.com/81074/153580095-242c3d16-4684-41ba-a88c-05e0cb08d831.mp4

# 最后

在微信读书中使用公众号可以降低打开微信的频次，如果你是微信读书的深度用户，还可以通过简悦将任意内容保存并完美的导入到微信读书中，同时使用 Vivaldi 的侧栏降低此方式的操作频次。

**Comment by felix2049lou** [comment]
想问如果在微信阅书标注的内容，会纪录在docx内？
